src/ucp/proto/proto_select.h

```diff
@@ -28,6 +28,10 @@
  * Relevant for UCP_OP_ID_RNDV_SEND and UCP_OP_ID_RNDV_RECV. */
 #define UCP_PROTO_SELECT_OP_FLAG_PPLN_FRAG (UCP_PROTO_SELECT_OP_FLAGS_BASE << 0)
 
+/* Restrict selection of rendezvous pipeline protocols (e.g. when sender buffer
+ * is non-contig). */
+#define UCP_PROTO_SELECT_OP_FLAG_NO_PPLN   (UCP_PROTO_SELECT_OP_FLAGS_BASE << 1)
```

Comment:
why need a special flag? can't rndv/ppln disqualify itself if the buffer is not contig according to select_param dt_class?

Reply:
fixed in place